src/app/selection/topics/page.tsx

```diff
@@ -1,14 +1,73 @@
 "use client";
 
-import { useSearchParams } from "next/navigation";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useState } from "react";
+import styles from "./topics.module.css";
+import { TOPIC_MAP, TopicKey } from "@/app/constants/topics";
+import Button from "@/app/common/Button/Button";
 
 export default function Topics() {
   const searchParams = useSearchParams();
-  const topic = searchParams.get("topic");
+  const router = useRouter();
+  const topic = searchParams.get("topic") as TopicKey;
+  const [selected, setSelected] = useState<string[]>([]);
+
+  const selectedTopics = topic ? TOPIC_MAP[topic] : {};
+  const allTopics = Object.keys(selectedTopics);
+  const isAllSelected = selected.length === allTopics.length;
+
+  const toggleSelectAll = () => setSelected(isAllSelected ? [] : allTopics);
+
+  const toggleSelectTopic = (item: string) => {
+    setSelected((prev) =>
+      prev.includes(item)
+        ? prev.filter((selectedItem) => selectedItem !== item)
+        : [...prev, item]
+    );
+  };
```

Comment:
```suggestion
const toggleSelectTopic = (item: string) => {
    if (selected.includes(item)) {
      setSelected(selected.filter((selectedItem) => selectedItem !== item));
      return;
    }

    setSelected([...selected, item]);
  };
```

너무 숏코딩하면 가독성이 안좋아서 보기 헷갈려요